### PR TITLE
assert_dom :text collapses whitespace

### LINKED ISSUE #121: `assert_dom` should ignore whitespace just like `assert_dom_equal`
In one of my views I have this:

```ruby
<li><%= item.quantity %> &times; <%= item.product.title %></li>
```

My formatter automatically breaks this onto new lines like so:

```ruby
      <li><%= item.quantity %>
        &times;
        <%= item.product.title %></li>
```

Because of this, running a test that contains this assertion fails:

```ruby
assert_select "ul li", "#{item.quantity} \u00D7 #{products(:one).title}"
```

This is due to the text I'm passing to `assert_select` having no newlines in it, but in the view code there _are_ newlines even though the browser actually drops those newlines.

So the test output is as follow:

```
Failure:
LineItemsControllerTest#test_should_create_line_item [test/controllers/line_items_controller_test.rb:27]:
--- expected
+++ actual
@@ -1 +1,3 @@
-"1 × The Pragmatic Programmer"
+"1
+        ×
+        The Pragmatic Programmer"
.
Expected 0 to be >= 1.
```

My current fix is to either turn off my formatter for that file, or to update the assertion to use regex and replace the spaces with `\s+` like this:

```ruby
assert_select "ul li", /#{item.quantity}\s+\u00D7\s+#{products(:one).title}/
```

However, the test should really treat excess whitespace the same as the browser does by dropping it.

Sounds like the same thing was done for `assert_dom_equal` in #84.

### PR DESCRIPTION
**Pull request by jyeharry**
- **Add macos to supported platforms in Gemfile.lock**
- **Add failing test for assert_dom collapsing whitespace**
- **Collapse whitespace from :text but not :html**

Fixes #121

This PR is my preferred solution over #122.

Instead of making use of a `:strict` operator to determine whether or not to collapse excess whitespace as the browser does, we instead use the existing `:html` equality operator to match for text with all whitespace included. The `:text` equality operator is then optimised to collapse all whitespace.

In my opinion, `:text` should match what would be returned by `Element.innerText` in javascript land, in that it does not include the excess whitespace. `Element.innerHTML` on the other hand, does include all of the whitespace that was in the document which I think is what the `:html` operator should match (which it does already) and so `:html` could instead be used as a `:strict` parameter.

The only difference between what `Element.innerText` and the updated `:text` equality operator does is that `Element.innerText` replaces `<br>` tags with `\n`, whereas our `:text` operator just removes them without replacing them. But this is out of our control as it is Nokogiri that does this. Regardless, this is my preferred solution but I made the two PRs since the original issue I raised specifically mentions a `strict` parameter.

### CODE REVIEW

**Comment by flavorjones** [comment]
There seems to be an issue with the CI workflows related to the Logger gem. I'll fix it on master and then you can rebase this and #122.

**Comment by jyeharry** [comment]
So to confirm, you won't _both_ PRs merged, not just this one? I made the two PRs so that there were a couple of options for how this would be implemented, with the intention that only one would get chosen. Let me know what you think is best.

I should probably also update the commented docs just before the `assert_dom` definition to make the updated behaviour of `:text` and `:html` a little more clear.

**Comment by flavorjones** [comment]
> So to confirm, you won't both PRs merged, not just this one

I just want whatever you want to submit rebased so I can see if the tests pass, please! Or let me know if you want me to do it, that's fine, too.

**Comment by jyeharry** [comment]
Ahhh right sorry for some reason I read "rebase" as merge the pull request. I'll rebase now.

**Comment by jyeharry** [comment]
@flavorjones Rebased and tests and ci is passing

**Comment by jyeharry** [comment]
@flavorjones just following up on this

**Comment by joelhawksley** [comment]
👋🏻 just wanted to share that this change broke the ViewComponent test suite: https://github.com/ViewComponent/view_component/actions/runs/15148835486/job/42590881883?pr=2309.

The fix was pretty minor (https://github.com/ViewComponent/view_component/pull/2309/files#diff-0553d15257b3cb95586c15820fe4dbe55ce1d4e3bae03f0f37d88fafcc50709b), but I would argue that this PR was a breaking change. 

I'm 💯 OK with leaving this be as the fix was very minor on our end, but figured I'd flag the issue in case others run into it.

**Comment by ccasabona** [comment]
Changing the behavior of collapsing the white space has broken hundreds of tests.  We'll have to stick with 2.2.0.

**Comment by jyeharry** [comment]
@rafaelfranca should some documentation or changelog be created to let users know that the update is a breaking change and that the `:html` equality operator must be used to match all whitespace?
